PowerPointLabs/PowerPointLabs/FunctionalTestInterface.Impl/Controller/TimerLabController.cs

```diff
@@ -0,0 +1,107 @@
+﻿using System;
+using System.Windows;
+using System.Windows.Controls.Primitives;
+
+using PowerPointLabs.ActionFramework.Common.Extension;
+using PowerPointLabs.TimerLab;
+using TestInterface;
+
+namespace PowerPointLabs.FunctionalTestInterface.Impl.Controller
+{
+    [Serializable]
+    class TimerLabController : MarshalByRefObject, ITimerLabController
+    {
+        private static ITimerLabController _instance = new TimerLabController();
+
+        public static ITimerLabController Instance { get { return _instance; } }
+
+        private TimerPane _pane;
+
+        private TimerLabController() {}
+
+        public void OpenPane()
+        {
+            UIThreadExecutor.Execute(() =>
+            {
+                FunctionalTestExtensions.GetRibbonUi().OnAction(
+                    new RibbonControl("TimerLabButton"));
+                _pane = FunctionalTestExtensions.GetTaskPane(
+                    typeof(TimerPane)).Control as TimerPane;
+            });
+        }
+
+        public void ClickCreateButton()
+        {
+            if (_pane != null)
+            {
+                UIThreadExecutor.Execute(() =>
+                {
+                    _pane.TimerPaneWPF.createButton.RaiseEvent(new RoutedEventArgs(ButtonBase.ClickEvent));
```

Comment:
Yes it doesnt support assignment. But still it can be done something like this `_pane?.setDurationTextBox(...)`.
Up to u, it just makes the codes cleaner. Other controller were written in older c#, so didnt use it